server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4852,9 +4860,58 @@ public UserVm createVirtualMachine(DeployVMCmd cmd) throws InsufficientCapacityE
                 _tmplService.attachIso(tmpl.getId(), vm.getId());
             }
         }
+
+        // Add extraConfig to user_vm_details table
+        Account caller = CallContext.current().getCallingAccount();
+        String extraConfig = cmd.getExtraConfig();
+        Long vmId = vm.getId();
+        if (!Strings.isNullOrEmpty(extraConfig)) {
```

Comment:
StringUtils.isNotBlank suggested as above. Also, please add the condition of the EnableAdditionalVmConfig global setting here and remove it from the addExtraConfig() method below

Reply:
Adapted change and moved global config setting